main/src/cgeo/geocaching/log/LogCacheActivity.java

```diff
@@ -85,6 +85,44 @@
 
     private static final int SELECT_IMAGE = 101;
 
+    private enum TrackableComparator {
+        TRACKABLE_COMPARATOR_TRACKCODE(0,R.string.trackable_code, new Comparator<TrackableLog>() {
+            @Override
+            public int compare(final TrackableLog lhs, final TrackableLog rhs) {
+                return TextUtils.COLLATOR.compare(lhs.trackCode, rhs.trackCode);
+            }
+        }),
+        TRACKABLE_COMPARATOR_NAME(1,R.string.trackable_name, new Comparator<TrackableLog>() {
+            @Override
+            public int compare(final TrackableLog lhs,  final TrackableLog rhs) {
+                return TextUtils.COLLATOR.compare(lhs.name, rhs.name);
+            }
+        });
+
+        private final int position;
+        private final int label;
+        private final Comparator<TrackableLog> comparator;
+        public static TrackableComparator[] comparators = new TrackableComparator[]
```

Comment:
I think this public mutable array can be omitted. 
Instead add a static method `TrackableComparator.findByPosition(final int position)`